docs/node/run-your-node/keymanager-node/key-manager-upgrade.md

```diff
@@ -0,0 +1,103 @@
+# Upgrading Key Managers
+
+This guide will describe how to upgrade a Key Manager node.
+
+## About the Upgrade
+
+Every Key Manager node contains all the keys used by confidential ParaTimes 
+inside its TEE-encrypted state. The key material is sealed and can only be 
+decrypted by the exactly same TEE enclave running on the exactly same CPU. This 
+means that newer Key Manager paratimes can not read the key material and that 
+key material can not be restored on another machine. 
+
+**During a Key Manager node upgrade it is therefore essential that the key 
+material is not lost, not even due to an operational error or even a 
+catastrophically failed upgrade.**
+
+## Safe Upgrade Procedure
+
+A Key Manager node's upgrade procedure differs from other Oasis nodes upgrades 
+because the upgraded node cannot unseal/decrypt the old Key Manager's state.
+
+To upgrade a Key Manager node, we need to delete the local state and let the 
+Key Manager's state replicate itself from other nodes. Only one Key Manager 
+runtime in the configuration file can be present at once. 
+
+**Always follow the safe upgrade procedure:**
+1. Keep the first half of nodes running on the old version
+2. Upgrade the second half
+3. Wait for the paratime upgrade epoch
+4. Verify that secrets have been replicated to the second half of nodes
+5. **Verify again**
+6. Upgrade the first half of the nodes
+
+### Upgrade Nodes
+
+To upgrade a Key Manager, follow the next steps:
+1. Wipe its local state `worker-local-storage.badger.db`
+```
+rm -rf runtimes/4000000000000000000000000000000000000000000000004a1a53dff2ae482d/worker-local-storage.badger.db/
+```
+2. Upgrade the Key Manager runtime
```

Comment:
Maybe explain what this means (e.g. get the new ORC, update the configuration to replace the ORC, restart the node).